x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/TimeSeriesLifecycleActionsIT.java

```diff
@@ -267,9 +270,36 @@ public void testWaitForSnapshotSlmExecutedBefore() throws Exception {
             }
         }, slmPolicy);
 
+        assertBusy(() -> {
+            Step.StepKey stepKey = getStepKeyForIndex(client(), index);
+            logger.info("stepKey for index {} is {}", index, stepKey);
+            assertThat(stepKey.getAction(), equalTo("complete"));
+        }, slmPolicy);
         assertBusy(() -> assertThat(getStepKeyForIndex(client(), index).getAction(), equalTo("complete")), slmPolicy);
     }
 
+    /*
+     * This method waits until phase_time gets set in the state store for the delete phase. Otherwise we can wind up starting a snapshot
+     * before the ILM policy is ready.
+     */
+    @SuppressWarnings("unchecked")
+    private void waitForPhaseTime() throws Exception {
+        assertBusy(() -> {
+            Request request = new Request("GET", "/_cluster/state/metadata/" + index);
```

Comment:
Just so you know (you don't have to change this if you don't want to), this information is available in `GET /<index>/_ilm/explain` which will show the ILM-related metadata, rather than getting it out of the cluster state.

Reply:
Any advantage to one way or the other?

Reply:
Generally using ILM explain would be better since these tests are specifically for ILM. That way if there were ever a disconnect in the ILM explain this would be more likely to catch it.